chirp/drivers/radtel_rt620.py

```diff
@@ -0,0 +1,545 @@
+            val = 0
+        else:
+            val = _mem.vfo.a.step
+        rs = RadioSetting("vfo.a.step", "Tuning Step",
+                          RadioSettingValueList(baofeng_uv17Pro.LIST_STEPS,
+                                                current_index=val))
+        vfoA.append(rs)
+        if _mem.vfo.b.step >= len(self.STEPS):
+            val = 0
+        else:
+            val = _mem.vfo.b.step
+        rs = RadioSetting("vfo.b.step", "Tuning Step",
+                          RadioSettingValueList(baofeng_uv17Pro.LIST_STEPS,
+                                                current_index=val))
+        vfoB.append(rs)
+        rs = RadioSetting("vfo.a.bcl", "Busy Channel Lock",
+                          RadioSettingValueBoolean(_mem.vfo.a.bcl))
+        vfoA.append(rs)
+        rs = RadioSetting("vfo.b.bcl", "Busy Channel Lock",
+                          RadioSettingValueBoolean(_mem.vfo.b.bcl))
+        vfoB.append(rs)
+        rs = RadioSetting("vfo.a.compander", "Compander",
+                          RadioSettingValueBoolean(_mem.vfo.a.compander))
+        vfoA.append(rs)
+        rs = RadioSetting("vfo.b.compander", "Compander",
+                          RadioSettingValueBoolean(_mem.vfo.b.compander))
+        vfoB.append(rs)
+        workmode.append(vfoA)
+        workmode.append(vfoB)
+
+    def _get_raw_memory(self, number):
+        if number == -1:
+            return self._memobj.vfo.b
+        elif number == -2:
+            return self._memobj.vfo.a
+        else:
+            return super()._get_raw_memory(number)
+
+    def get_memory(self, number):
+        #        if isinstance(number, str):
+        #            number = self._get_specials()[number]
+        mem = super().get_memory(number)
+#          mem.number = name
+        _mem = self._get_raw_memory(number)
+#        if _mem.rxtone == 0:
+#            _mem.rxtone = 0xFFFF
+#        if _mem.txtone == 0:
+#            _mem.txtone = 0xFFFF
+        if hasattr(mem.extra, "keys") and \
+                "compander" not in mem.extra.keys():
+            rs = RadioSetting("compander", "Compander",
+                              RadioSettingValueBoolean(_mem.compander))
+            mem.extra.append(rs)
+        mem.tuning_step = self.STEPS[_mem.step] or self.STEPS[0]
+        if _mem.wide < len(self.MODES):
+            mem.mode = self.MODES[_mem.wide]
+#       if isinstance(mem.number, str):
+#           mem.vfo=1
+        return mem
+
+    def set_memory(self, mem):
+        _mem = self._get_raw_memory(mem.number)
+        super().set_memory(mem)
+        _mem.step = self.STEPS.index(mem.tuning_step)
+        _mem.wide = self.MODES.index(mem.mode)
+        if _mem.rxtone == 0:
+            _mem.rxtone = 0xFFFF
+        if _mem.txtone == 0:
+            _mem.txtone = 0xFFFF
+
+    def get_features(self):
+        rf = super().get_features()
+        rf.has_tuning_step = True
+        rf.valid_tuning_steps = self.STEPS
+        rf.valid_modes = self.MODES
```

Comment:
This is already done on the base class so you should not need to do it again here.

Reply:
As you are mentioning it for `rf.valid_modes = self.MODES` , it's true, the base class should use my values. Is it possible the base class was refactored in last month or so and originally both STEPS and MODES were defined as global variables in the original class (with STEPS staying that way for now)? Because everything which was in base class I have tried to use directly (but I could have overlooked this).